test/models/metadata.test.ts

```diff
@@ -1,11 +1,84 @@
 import { expect } from '@open-wc/testing';
 
 import { Metadata } from '../../src/models/metadata';
+import { DateParser } from '@internetarchive/field-parsers';
+import { DateField } from '../../src/models/metadata-fields/field-types/date';
+
+// Grab the names of all the getter fields on Metadata objects
+// (with the exception of `identifier`, which is just a raw string rather
+// than a field-type like the others).
+const fieldNames: (keyof Metadata)[] = (Object.keys(
+  Object.getOwnPropertyDescriptors(Metadata.prototype)
+) as (keyof Metadata)[]).filter(
+  field => !['constructor', 'identifier'].includes(field)
+);
 
 describe('Metadata', () => {
   it('properly instantiates metadata with identifier', async () => {
     const json = { identifier: 'foo', collection: 'bar' };
     const metadata = new Metadata(json);
     expect(metadata.identifier).to.equal('foo');
+    expect(metadata.collection?.value).to.equal('bar');
+  });
+
+  it('handles missing data gracefully', () => {
```

Comment:
(Not strictly related, but I noticed that the Metadata class was lacking proper unit tests so figured I'd add a couple)